### PR TITLE
Bump protobuf and absl.

### PR DESCRIPTION
**Pull request by lhruby**
# Description
Bump protobuf to v27.2 and absl to the matching 20240116.2. See absl release notes for breaking changes. 

Bump `DEPS_VERSION=1.0.17` accordingly.

## Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] If this is a new component I have added examples.
- [ ] I updated the README and related documentation.

### CODE REVIEW

**Review by filippobrizzi — APPROVED** [review]

File: external/CMakeLists.txt [hunk 1]
```diff
@@ -46,8 +46,8 @@ add_cmake_dependency(
 
 # --------------------------------------------------------------------------------------------------
 # Abseil C++
-set(ABSEIL_VERSION 20230802)
-set(ABSEIL_TAG ${ABSEIL_VERSION}.1)
+set(ABSEIL_VERSION 20240116)
+set(ABSEIL_TAG ${ABSEIL_VERSION}.0)
```
Comment (floriantschopp) [inline external/CMakeLists.txt:50]:
```suggestion
set(ABSEIL_TAG ${ABSEIL_VERSION}.2)
```

File: external/CMakeLists.txt [hunk 2]
```diff
@@ -61,8 +61,8 @@ add_cmake_dependency(
 
 # --------------------------------------------------------------------------------------------------
 # protobuf
-set(PROTOBUF_TAG v25.2)
-set(PROTOBUF_VERSION 4.25.2)
+set(PROTOBUF_TAG v27.2)
+set(PROTOBUF_VERSION 4.27.2)
```
Comment (floriantschopp) [inline external/CMakeLists.txt:65]:
```suggestion
set(PROTOBUF_VERSION 5.27.2)
```
I think is correct here...

Comment (lhruby) [reply external/CMakeLists.txt]: great catch, changed

File: external/CMakeLists.txt (hunk 1 above)
Comment (lhruby) [reply external/CMakeLists.txt]: done

**Review by floriantschopp — APPROVED** [review]

**Pull request merged**